eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -531,17 +531,36 @@ class EclairImpl(appKit: Kit) extends Eclair with Logging {
   }
 
   private def getChannelDescs(shortChannelIds: Set[ShortChannelId])(implicit timeout: Timeout): Future[Set[ChannelDesc]] = {
+    def toChannelDesc(shortChannelId: ShortChannelId, nodeId1: PublicKey, nodeId2: PublicKey): Set[ChannelDesc] = {
+      Set(
+        ChannelDesc(shortChannelId, nodeId1, nodeId2),
+        ChannelDesc(shortChannelId, nodeId2, nodeId1))
+    }
+
+    def getLocalChannelDescs: Future[(Set[ChannelDesc], Set[ShortChannelId])] = {
+      channelsInfo(None).map { channels =>
+        channels.foldLeft((Set.empty[ChannelDesc], shortChannelIds)) { (acc, channel) =>
+          channel.data match {
+            case data: DATA_NORMAL if shortChannelIds(data.channelUpdate.shortChannelId) =>
+              val descs = toChannelDesc(data.channelUpdate.shortChannelId, channel.nodeId, appKit.nodeParams.nodeId)
+              val (channelDescs, nonLocalChannels) = acc
+              (channelDescs ++ descs, nonLocalChannels - data.channelUpdate.shortChannelId)
+            case _ => acc
+          }
+        }
+      }
+    }
+
     if (shortChannelIds.isEmpty) {
       Future.successful(Set.empty)
     } else {
       for {
+        (localChannelDescs, nonLocalChannelIds) <- getLocalChannelDescs
         channelsMap <- (appKit.router ? GetChannelsMap).mapTo[SortedMap[ShortChannelId, PublicChannel]]
```

Comment:
I believe a much simpler fix would be to do `(appKit.router ? GetRouterData)` and then use `d.resolve` which will look at both private and public channels and correctly handle scid aliases. This would require only a few lines of code and should be more future-proof.

Reply:
I don't understand what you did in https://github.com/ACINQ/eclair/pull/2483/commits/92c416986e08aa59f0b49e9474527cb6cabc547f

Once you cal `GetRouterData`, you don't need `getLocalChannelDescs` at all anymore, you have all the data necessary and can remove most of your added code?